### DISCUSSION TITLE
Don't allow only certain sites

### DISCUSSION THREAD

**Discussion by Elektrot** [RaspAP/raspap-webgui#744] [General]
Hi,
How are links only to certain sites or ip addresses allowed?
Thank you.

**Comment by billz** [comment]
This is usually called host or IP whitelisting (the opposite of RaspAP's ad block feature). There are many ways to achieve this, but the simplest method I would suggest is by using dnsmasq. Edit the file `/etc/dnsmasq.d/090_raspap.conf` and add the following:

```
address=/#/127.0.0.1
server=/some-website.com/9.9.9.9
```

The `address` line tells dnsmasq to redirect all sites to localhost. The `server` directive makes only certain sites use Quad9's DNS server at 9.9.9.9. You can add any number of these `server` lines to build a custom whitelist.

Restart dnsmasq with RaspAP or from the shell and check its status:

```
sudo systemctl restart dnsmasq.service
sudo systemctl status dnsmasq.service
```

**Reply by Elektrot** [reply]
**Thank you so much**

**Reply by K4TFJ** [reply]
Is this still valid in 2024? I attempted this once already and it did not seem to work.  I will try again. Just for clarification, I need to block all traffic on the AP (wlan1 in my case) except for 3 domains. I really do hope this is possible. 

Additional testing and details:

RPi4b/4B with 32GB SDC, 
RPiOS 64 Lite (Bookworm)
USB Wireless 802.11ac

Added this (with the correct domain) to the /etc/dnsmasq.d/090_raspap.conf file:

	address=/#/127.0.0.1
	server=/some-website.com/9.9.9.9

performed the following:
	sudo systemctl restart dnsmasq.service
	sudo systemctl status dnsmasq.service
	
no change was noticed on a device connect to AP. Many websites (ie google.com) available.

Rebooted and reconnected wifi-client (Internet upstream)
device reconnected to AP

tried this also with /etc/dnsmasq.d/090_wlan1.conf
	
same results.

**Reply by billz** [reply]
Best advice is to consult the dnsmasq docs.
